src/linear_proof.rs

```diff
@@ -165,7 +168,12 @@ impl LinearProof {
         b_vec: Vec<Scalar>,
     ) -> Result<(), ProofError> {
         transcript.innerproduct_domain_sep(n as u64);
+        assert_eq!(b_vec.len(), n);
+
         transcript.append_point(b"C", &C);
+        for i in 0..n {
+            transcript.append_scalar(b"b_i", &b_vec[i]);
+        }
```

Comment:
The added assertion ensures that the indexing operation won't panic, but I would personally prefer:
```suggestion
        for b_i in &b_vec {
            transcript.append_scalar(b"b_i", b_i);
        }
```